spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/KafkaBinderMetricsTest.java

```diff
@@ -103,6 +104,30 @@ public void shouldIndicateLag() {
 						.isEqualTo(500.0);
 	}
 
+	@Test
+	public void shouldNotContainAnyMetricsWhenUsingNoopGauge() {
+		// Adding NoopGague for the offset metric.
+		meterRegistry.config().meterFilter(
+				MeterFilter.denyNameStartsWith("spring.cloud.stream.binder.kafka.offset"));
+
+		// Because we have NoopGauge for the offset metric  in the meter registry, none of these expectations matter.
+		org.mockito.BDDMockito
+				.given(consumer.committed(ArgumentMatchers.any(TopicPartition.class)))
+				.willReturn(new OffsetAndMetadata(500));
+		List<PartitionInfo> partitions = partitions(new Node(0, null, 0));
+		topicsInUse.put(TEST_TOPIC,
+				new TopicInformation("group1-metrics", partitions, false));
+		org.mockito.BDDMockito.given(consumer.partitionsFor(TEST_TOPIC))
+				.willReturn(partitions);
+		metrics.bindTo(meterRegistry);
+
+		// Because of the NoopGause, the meterRegistry should contain no metric.
+		assertThat(meterRegistry.getMeters()).hasSize(0);
+
+		// Reset the MeterRegistry to accept all metrics.
+		meterRegistry.config().meterFilter(MeterFilter.accept());
```

Comment:
This doesn't reset anything, it just adds the accept all filter at the end; you will need to use reflection (e.g. `KafkaTestUtils.setPropertyValue()`) to reset the filter array to `new MeterFilter[0]`.